src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -124,12 +125,29 @@ export const monoPlatform: Platform = {
       return unboxedValue;
     }
 
-    return BINDING.conv_string(fieldValue as any as System_String);
+    let decodedString: string | null | undefined;
+    if (currentHeapLock) {
+      decodedString = currentHeapLock.stringCache.get(fieldValue);
+      if (decodedString === undefined) {
+        decodedString = BINDING.conv_string(fieldValue as any as System_String);
+        currentHeapLock.stringCache.set(fieldValue, decodedString);
+      }
+    } else {
+      decodedString = BINDING.conv_string(fieldValue as any as System_String);
```

Comment:
Is there code that hits this code path?

Reply:
Yes, any of the framework's built-in unmarshalled interop calls that read strings would hit this.

Reply:
I know you could argue that they too should take a heap lock, but I'm not trying to solve all problems here.